types/svgo/index.d.ts

```diff
@@ -568,11 +569,64 @@ export type SortAttrsPlugin = DefaultPlugin<
 >;
 
 /**
- * Sorts children of <defs> to improve compression
+ * sorts children of <defs> to improve compression
  */
 export type SortDefsChildrenPlugin = DefaultPlugin<'sortDefsChildren'>;
 
+export interface Preset<N extends string, P = never> {
+    name: N;
+    params?: P | undefined;
+}
+
+/**
+ * plugins which are enabled in default preset
+ */
+export type DefaultPresetPlugins =
```

Comment:
as you probably noted:
```ts
    | RemoveAttrsPlugin
    | RemoveAttrsPlugin
```
there is dev-ex problem here. What about using those types you've added and existng ones to type which is union of literals? To avoid duplicates in both types

Reply:
@peterblazejewicz I don't fully understand what you mean by that, so I will try to describe the purpose of these changes.

There are two unions, because there are two different concepts in the svgo library.

First you can activate plugins via name or via full object configuration:
```
plugins: [
        'addAttributesToSVGElement',
        { name: 'cleanupAttrs', params: { newlines: true, trim: true, spaces: true } },
]
```

Second there is a new way to configure the default (preloaded) plugins via the special `'preset-default'` plugin, which is actually not a plugin because it has a different type signature. Inside this `'preset-default'` plugin, one can configure all the default-loaded plugins (this is what the second union is for).
```
{
            name: 'preset-default',
            params: {
                floatPrecision: 4,
                overrides: {
                    cleanupAttrs: { newlines: true, trim: true, spaces: true },
                    cleanupEnableBackground: false,
           }
}
```

The only thing which I don't really like is the difference between these two configuration ways. One uses plugin names as strings, the other as object keys. This makes it complicated to write clean types. I chose to use the existing types and just write `{ [P in DefaultPresetPlugins['name']]?: false | DefaultPresetPlugins['params']}` which is very ugly and unions all the type options between all plugins instead of properly hinting the devs: this plugin has only this set of options.

Reply:
oh, this could be that sounded cryptic. We have those literals, and they are reused in both union types. I would support, if there is purpose, to reuse subset as union with other literals, to save typing and duplication, say `type A = ..., type B = A | ....;`. If there is a gain ofc.
sorry if I was not expressing myself clearly

Reply:
Is this what you meant?
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/55226/commits/00f4175232d36ac867bdd7c29408819006d4442e

Reply:
Yes, thx!